dotnet-desktop-guide/net/winforms/data/ensure-the-selected-row-in-a-child-table-correct.md

```diff
@@ -0,0 +1,60 @@
+---
+title: "How to ensure the selected row in a child table remains at the correct position"
+description: "Learn about how to ensure the selected row in a child table remains at the correct position."
+ms.date: "06/08/2022"
+dev_langs:
+  - "csharp"
+  - "vb"
+helpviewer_keywords: 
+  - "master-details view"
+  - "row position [Windows Forms .NET]"
+  - "parent/child view [Windows Forms .NET]"
+  - "resetting child position"
+  - "data binding [.NET], row selection"
+  - "caching [.NET], child position"
+  - "child position"
+  - "master/details view [Windows Forms .NET]"
+  - "child tables row selection"
+  - "current child position"
+ms.custom: devdivchpfy22
+---
+
+# How to ensure the selected row in a child table remains at the correct position in Windows Forms .NET
+
+Often when you work with data binding in Windows Forms, you'll display data in a parent/child or master/details view. The data from the same source is displayed in two controls and is referred to as data-binding. Changing the selection in one control causes the data displayed in the second control to change. For example, the first control might contain a list of customers and the second a list of orders related to the selected customer in the first control.
+
+ To display data in a parent/child view, you might have to take extra steps to ensure that the currently selected row in the child table isn't reset to the first row of the table. You'll have to cache the child table position and reset it after the parent table changes to view the currently selected row in the child table. Typically the child reset occurs the first time a field in a row of the parent table changes.
```

Comment:
Considering this article is talking about the `CurrencyManager` type, I think we should actually introduce it. Also, this paragraph is indented by one space. Please check and correct all indentations in this article.

```suggestion
To display data in a parent/child view, you might have to take extra steps to ensure that the currently selected row in the child table isn't reset to the first row of the table. You'll have to cache the child table position and reset it after the parent table changes to view the currently selected row in the child table. Typically the child reset occurs the first time a field in a row of the parent table changes.

## Currency manager

The <xref:System.Windows.Forms.BindingSource> type encapsulates a data source for the controls on a form. Navigating the data in the binding source is the responsibility of a binding manager. When the data is an <xref:System.Collections.IList> collection, that binding manager is a <xref:System.Windows.Forms.CurrencyManager>. Likewise, when one of the items of the data is also an `IList`, such as a property that is a list of house numbers associated with a street, you must obtain the `CurrencyManager` for that item to navigate that subset of data. Use the <xref:System.Windows.Forms.BindingSource.GetRelatedCurrencyManager%2A?displayProperty=nameWithType> method to retrieve the currency manager for the item.

The currency manager provides information about the data source list, such as the count of items and the index of the selected item. The currency manager also provides events to help notify you about when the item changes, such as when a property of the item is altered, when the list of items is changed, and when an item is selected.
```